src/unix/core.c

```diff
@@ -998,6 +998,41 @@ int uv_getrusage(uv_rusage_t* rusage) {
   return 0;
 }
 
+int uv_getrusage_thread(uv_rusage_t* rusage) {
+#ifndef RUSAGE_THREAD
+  return UV_ENOTSUP;
+#else
+  struct rusage usage;
+
+  if (getrusage(RUSAGE_THREAD, &usage))
+    return UV__ERR(errno);
+
+  rusage->ru_utime.tv_sec = usage.ru_utime.tv_sec;
+  rusage->ru_utime.tv_usec = usage.ru_utime.tv_usec;
+
+  rusage->ru_stime.tv_sec = usage.ru_stime.tv_sec;
+  rusage->ru_stime.tv_usec = usage.ru_stime.tv_usec;
+
+#if !defined(__MVS__) && !defined(__HAIKU__)
```

Comment:
Do you think we should initialize the `uv_rusage_t` struct to zero `memset(rusage, 0, sizeof(*rusage))` so the fields not supported in some platforms are set to 0?

Reply:
The current implementation of uv_getrusage does not do so. I'd rather keep it consistent.

Reply:
It seems more like on oversight to me and I think consistency between unices and windows wrt unsupported fields is a good thing to have. Regardless, this can be dealt with in a subsequent PR.

Reply:
Ok. I also noticed there's no NULL check guard in the existing function. Should there be?